src/v/cloud_io/io_result.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "cloud_io/io_result.h"
+
+namespace cloud_storage {
```

Comment:
namespace should probably also be `cloud_io`

Reply:
Yea I kept it here for now to avoid a bunch of code churn. Will defer to a follow-up if you don't mind